src/common.jl

```diff
@@ -1,194 +1,267 @@
+    k = constant_term.(duT) # For the interpolation, needs k at the updated point
+    integrator.destats.nf += 1
 end
 
-# This function is part of OrdinaryDiffEq.jl; MIT-licensed
-# https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/8b5af68ee4126d3772fe690a7128468f1334b4d6/src/solve.jl#L411
-function tstop_saveat_disc_handling(tstops, saveat, d_discontinuities, tspan)
-    t0, tf = tspan
-    tType = eltype(tspan)
-    tdir = sign(tf - t0)
+stepsize_controller!(integrator,alg::_TaylorMethod) = stepsize(integrator.cache.uT, integrator.opts.abstol)
+step_accept_controller!(integrator, alg::_TaylorMethod, q) = q
 
-    tdir_t0 = tdir * t0
-    tdir_tf = tdir * tf
+function DiffEqBase.solve(
+        prob::DiffEqBase.AbstractODEProblem{uType, tupType, isinplace},
+        alg::TaylorMethod, args...;
+        verbose=true,
+        kwargs...) where
+        {uType, tupType, isinplace}
 
-    # time stops
-    tstops_internal = BinaryMinHeap{tType}()
-    if isempty(d_discontinuities) && isempty(tstops) # TODO: Specialize more
-      push!(tstops_internal, tdir_tf)
+    if verbose
+        warned = !isempty(kwargs) && check_keywords(alg, kwargs, warnlist)
+        warned && warn_compat()
+    end
+
+    f = prob.f
+    if !isinplace && typeof(prob.u0) <: AbstractArray
+        if prob.f isa DynamicalODEFunction
+            f1! = (dv, v, u, p, t) -> (dv .= prob.f.f1(v, u, p, t); 0)
+            f2! = (du, v, u, p, t) -> (du .= prob.f.f2(v, u, p, t); 0)
+            _alg = _TaylorMethod(alg.order, parse_eqs = false)
+            ### workaround use of `SVector` with oop `DynamicalODEProblem`
+            ### TODO: add proper support for oop problems with arrays
+            if eltype(prob.u0.x) <: SVector
+                _u0 = ArrayPartition(SizedArray{Tuple{length(prob.u0.x[1])}}.(prob.u0.x))
+            else
+                _u0 = prob.u0
+            end
+            _prob = DynamicalODEProblem(f1!, f2!, _u0.x[1], _u0.x[2], prob.tspan, prob.p; prob.kwargs...)
+        else
+            f! = (du, u, p, t) -> (du .= f(u, p, t); 0)
+            _alg = _TaylorMethod(alg.order, parse_eqs = false)
+            _prob = ODEProblem(f!, prob.u0, prob.tspan, prob.p; prob.kwargs...)
+        end
+    elseif haskey(kwargs, :parse_eqs)
+        _alg = _TaylorMethod(alg.order, parse_eqs = kwargs[:parse_eqs])
+        _prob = prob
     else
-      for t in tstops
-        tdir_t = tdir * t
-        tdir_t0 < tdir_t ≤ tdir_tf && push!(tstops_internal, tdir_t)
-      end
+        _alg = _TaylorMethod(alg.order)
+        _prob = prob
+    end
 
-      for t in d_discontinuities
-        tdir_t = tdir * t
-        tdir_t0 < tdir_t ≤ tdir_tf && push!(tstops_internal, tdir_t)
-      end
+    # DiffEqBase.solve(prob, _alg, args...; kwargs...)
+    integrator = DiffEqBase.__init(_prob, _alg, args...; kwargs...)
+    integrator.dt = stepsize(integrator.cache.uT, integrator.opts.abstol) # override handle_dt! setting of initial dt
```

Comment:
Previously, there was a default value set for `abstol`; I cannot see it now. So the question is if the user must now to specify it, or is it set somewhere else?

Reply:
Good catch! If the user doesn't specify `abstol` it is set automatically by `DiffEqBase.__init` to `abstol=1e-6`